File: network_tcp_client_server/python_network_tcp_client_server/network_tcp_server.py
#comment
import sys
import socket
#import getopt

def server_driver():
  serversocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
  serversocket.bind((socket.gethostname(), 9876))
  serversocket.listen(1)
  #data,addr = UDPSock.recvfrom(buf)
  #serversocket.recvfrom(buf)
  #print buf
  print "TCPServer Waiting for client on port 9876."
  client_socket, address = serversocket.accept()
  data = client_socket.recv(1024)
  #print "server: received message=" . data
  print data
  client_socket.send("ok.")
  client_socket.close()
  serversocket.close()

def main():
  if len(sys.argv) != 1:
    print "Usage: %s <noargs>" % sys.argv[0]
    sys.exit()
  #print(sys.argv[1])
  server_driver()
  sys.exit(), end

main()

File: network_tcp_client_server/csharp_network_tcp_client_server/network_tcp_server.cs
using System;
using System.Text;
using System.Net;
using System.Net.Sockets;

public class NetworkTcpServer {
  public static void Main( string[] args ) {
    if( args.Length != 0 ) {
      Console.WriteLine("Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>");
      Environment.Exit(1);
    }

    uint SERVER_PORT = 9876;
    server_driver(SERVER_PORT);
  }

  public static void server_driver( uint port ) {
    IPHostEntry ipHostInfo = null;
    IPAddress ipAddress = null;
    Socket socketListener = null;
    Socket socketClient = null;
    byte[] bytes = null;
    string message = string.Empty;

    ipHostInfo = Dns.GetHostEntry(Dns.GetHostName());
    //ipHostInfo = Dns.GetHostByName(Dns.GetHostName());
    ipAddress = ipHostInfo.AddressList[0];
    socketListener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
    socketListener.Bind( new IPEndPoint(ipAddress, (int)port));
    socketListener.Listen(1);     // listen for 1 connection

    Console.WriteLine("The server is listening with name " + Dns.GetHostName() + " : " + port);
    socketClient = socketListener.Accept();
    while( socketClient.Available == 0 ) {
    }
    bytes = new byte[socketClient.Available];
    if( socketClient.Receive(bytes) > 0 ) {
      Console.WriteLine("server: received message='" + Encoding.ASCII.GetString(bytes) + "'");
    }

    message = "** Done. **";
    socketClient.Send(Encoding.ASCII.GetBytes(message));
    Console.WriteLine("server: send message='" + message + "'");
    socketClient.Close();
    Console.WriteLine("Complete.");
    Console.ReadLine();
  }
}
